firefox-bootstrap/bootstrap.js

```diff
@@ -0,0 +1,259 @@
+          "nsISupportsWeakReference"]),
+        onStateChange: function(webProgress, request, flag, status) {
+          if (flag & Ci.nsIWebProgressListener.STATE_STOP) {
+            if (request.originalURI && request.originalURI.host === WEBCOMPAT_DOMAIN) {
+              let mm = browser.selectedBrowser.messageManager;
+              browser.removeProgressListener(this);
+
+              // did we successfully get a screenshot?
+              if (tabData && tabData.canvasData) {
+                mm.loadFrameScript(FRAMESCRIPT, true);
+                mm.sendAsyncMessage(SCREENSHOT_MESSAGE, {
+                  screenshot: tabData.canvasData
+                });
+
+                mm.addMessageListener(ACK_MESSAGE, function ackFn(message) {
+                  mm.removeMessageListener(ACK_MESSAGE, ackFn);
+                  resolve();
+                });
+              } else {
+                resolve();
+              }
+              httpRequestObserver.removeObserver();
+            }
+          }
+        }
+      };
+
+      httpRequestObserver.addObserver();
+      browser.addProgressListener(loadedListener);
+      browser.loadOneTab(
+        `https://${WEBCOMPAT_DOMAIN}/issues/new?url=${encodeURIComponent(tabData.url)}`,
+        {inBackground: false});
+    });
+  },
+
+  getScreenshot(browser) {
+    const FRAMESCRIPT = `chrome://webcompat-reporter/content/tab-frame.js?${Date.now()}`;
+    const TABDATA_MESSAGE = "WebCompat:SendTabData";
+
+    return new Promise((resolve) => {
+      let mm = browser.selectedBrowser.messageManager;
+
+      mm.loadFrameScript(FRAMESCRIPT, true);
+      mm.addMessageListener(TABDATA_MESSAGE, function receiveFn(message) {
+        mm.removeMessageListener(TABDATA_MESSAGE, receiveFn);
+        resolve([browser, message.json]);
+      });
+    });
+  },
+
+  reportIssue(xulDoc) {
+    Promise.resolve(xulDoc.defaultView.gBrowser).then(this.getScreenshot)
+                                                .then(this.openWebCompatTab)
+                                                .catch(Cu.reportError);
+  }
+};
+
+// use enabled pref as a way for tests (e.g. test_contextmenu.html) to disable
+// the addon when running.
+let prefObserver = function(aSubject, aTopic, aData) {
+  let enabled = Services.prefs.getBoolPref(PREF_WC_REPORTER_ENABLED);
+  if (enabled) {
+    WebCompatReporter.init();
+  } else {
+    WebCompatReporter.uninit();
+  }
+};
+
+function startup(aData, aReason) { // eslint-disable-line no-unused-vars
+  // Observe pref changes and enable/disable as necessary.
+  Services.prefs.addObserver(PREF_WC_REPORTER_ENABLED, prefObserver, false);
+
+  // temporary for add-on port, set this in all.js somewhere, probably
+  Services.prefs.setBoolPref(PREF_WC_REPORTER_ENABLED, true);
+  // Only initialize if pref is enabled.
+  let enabled = Services.prefs.getBoolPref(PREF_WC_REPORTER_ENABLED);
+  if (enabled) {
+    WebCompatReporter.init();
```

Comment:
oops, forgot you need to click on a link for that to work in nightly or devedition: https://miketaylr.com/tmp/reporter.html